compose/config/config.py

```diff
@@ -718,7 +732,7 @@ def process_service(service_config):
             for path in to_list(service_dict['env_file'])
         ]
 
-    if 'build' in service_dict:
+    if 'build' in service_dict and 'build' not in skipped_sections:
```

Comment:
Most of the design for the feature is made in a generic fashion, but we end up just making a couple string matches (here and in `validate_paths`) which seems to defeat the purpose. What's the rationale for not removing the declared `skipped_sections` from the config entirely? Wouldn't it make #1973 simpler to implement down the line?

Reply:
Thanks for the review.

Not much of a rationale except I didn't think about it and minimum entropy production principle...

Now that you mention it, I'm mostly fine with it except I'm not sure if we should change original config that downward components might need.

Pretty busy this week. Will see if I can update the PR if I get another sleepless night.